FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx

```diff
@@ -320,20 +342,25 @@ void vtkPointMatcher::UpdateBestMatchingForSubsetOfPoints( vtkPoints* pointSubse
     permutedPointSubset2DistanceMatrix->Update();
     double rootMeanSquareDistanceErrorMm = this->ComputeRootMeanSquareistanceErrors( pointSubset1DistanceMatrix, permutedPointSubset2DistanceMatrix );
 
-    // TODO
-    // if suitability within some threshold of the best
-    //   flag ambiguous
+    // flag ambiguous if suitability within some threshold of the best result so far
+    double differenceComparedToBestMm = this->ComputedRootMeanSquareDistanceErrorMm - rootMeanSquareDistanceErrorMm;
+    if ( fabs( differenceComparedToBestMm ) <= this->AmbiguityThresholdDistanceMm )
+    {
+      this->MatchingAmbiguous = true;
```

Comment:
Is IsMatchingAmbiguous variable guaranteed to be set in all cases?

Reply:
What do you mean by "guaranteed to be set"? What do you mean by "all cases"?

Reply:
Just make sure that you always assign a value to this->MatchingAmbiguous. Currently, it is not obvious that it is set in all cases, only when you get inside one of the if statements.